CleverRaven/Cataclysm-DDA
# Changing movement mode to walking after waking up

#### Summary
None


#### Purpose of change
Waking up after sleeping leaves the player prone, this PR changes their movement mode to walking after waking up. Fixes #50644 



#### Describe the solution

So I've added the change to movement mode if you cancel the sleep activity voluntarily due to the tossing and turning prompt.

I also added a start walking if prone to the character wake up function, but this has some issues below.

* It currently says "You start Walking" and then "You Wake Up" I think this because the remove laying_down effect gets printed after the wake_up function is called, while the  movement mode change happens in wake_up., see attached screenshot
    * If there was a way to standup as part of removing the effect that would be best, I'm not sure how to do that
    * If we did that we should also put going prone as part of starting the effect
 


#### Describe alternatives you've considered

Using an effect to handling going into the walk movement mode, but I'm not sure if that's possible.



#### Testing

For failing to sleep during the sleep activity:

1. Make an evacuee character
 2. Sleep
 3. Select the prompt when it appears to stop trying to sleep
 4. The character starts walking

For calling wake_up (actually having the sleep effect)

1. Make an evacuee character
2. Spawn blindfold, ear plugs, fur sleeping bag, and 3 poppy cough syrup
3. Place the sleeping bag on the ground
4. Drink a bunch of poppy cough syrup (like 12 doses)
5. Put in ear plugs and blindfold
6. Sleep and wake up naturally
7. The character starts walking



#### Additional context
